### PR TITLE
Categorize jobs and add PM roles

### PR DESCRIPTION
**Pull request by timsneath**
_Description of what this PR is changing or adding, and why:_

- Adds headers to the jobs page to make it easy to scan engineering / devrel / PM&UX roles. 
- Adds two PM roles that we're hiring for. 

_Issues fixed by this PR (if any):_

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/jobs/pm_devexp.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+Here on the Flutter team and at Google, we embrace our differences
+and are [committed to furthering our culture of inclusion](https://flutter.dev/culture).
+In addition to groups like the [Flutteristas](https://flutteristas.org/),
+[Employee Resource Groups (ERGs)](https://diversity.google/commitments/)
+are employee-initiated networks for supporting underrepresented employees
+and their allies with shared values of creating belonging 
+across their communities and Google.
+
+### Work-Life Balance
```
Comment (sfshaza2) [inline src/jobs/pm_devexp.md:52]:
```suggestion
### Work-Life balance
```

File: src/jobs/pm_devexp.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+---
+title: Product Manager, iOS and Design Agencies
```
Comment (sfshaza2) [inline src/jobs/pm_devexp.md:2]:
```suggestion
title: Product Manager, Flutter Developer Experience
```

**Review by sfshaza2 — APPROVED** [review]
lgtm with a couple edits.

**Comment by timsneath** [comment]
Thank you for saving me from myself, @sfshaza2! :) 

BTW, we should fix the capitalization on `_template.md`, since that's where I took that section.

**Pull request merged**